pwiz_tools/Skyline/Model/SmallMoleculeTransitionListReader.cs

```diff
@@ -1011,6 +1009,18 @@ private void GetMoleculeDetails()
             }
         }
 
+        private void ThrowIfNotParserError(Exception exception)
+        {
+            if (exception is InvalidOperationException
+                || exception is InvalidDataException
+                || exception is ArgumentException)
+            {
+                return;
+            }
+
+            throw exception;

```

Comment:
This overwrites the stack of the original exception. Not a good idea. These are only 3 types of exceptions. Better to catch them all and be slightly redundant than to do this and blow away the stack of a programming error exception.